server/routes/incidentDetails/incidentDetailsSubmittedEdit.ts

```diff
@@ -62,39 +108,110 @@ export default class IncidentDetailsSubmittedEditRoutes {
       locations,
       data,
       reportingOfficer: reporter || '',
-      submitButtonText: 'Continue',
-      cancelButtonHref: cancelButtonHref || '/place-a-prisoner-on-report',
-      reportPreviouslySubmitted: true,
+      submitButtonText: 'Save and continue',
+      exitButtonHref: exitButtonHref || '/place-a-prisoner-on-report',
       adjudicationNumber: adjudicationDetails.adjudicationNumber,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
     })
   }
 
-  view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
+  view = async (req: Request, res: Response): Promise<void> => {
+    const pageData = {
+      incidentDate: req.session.incidentDate,
+      locationId: req.session.incidentLocation as unknown as string,
+      originalRadioSelection: req.session.originalRadioSelection,
+    }
+    delete req.session.incidentDate
+    delete req.session.incidentLocation
+    return this.renderView(req, res, pageData)
+  }
 
   submit = async (req: Request, res: Response): Promise<void> => {
-    const { incidentDate, locationId } = req.body
+    const {
+      incidentDate,
+      locationId,
+      currentRadioSelected,
+      search,
+      deleteAssociatedPrisoner,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
+    } = req.body
     const { user } = res.locals
     const { prisonerNumber, id } = req.params
+    const selectedPerson = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
+    const { originalRadioSelection } = req.session
     const { referrer } = req.query
 
-    const error = validateForm({ incidentDate, locationId })
-    if (error) return this.renderView(req, res, { error, incidentDate, locationId })
+    if (deleteAssociatedPrisoner) {
```

Comment:
We need to do something about this duplicate code - it is in 2 or 3 places in almost exactly the same form, and has the same reason for change.